packages/core/src/components/DeckSwiper/DeckSwiper.tsx

```diff
@@ -23,145 +27,171 @@ export interface DeckSwiperProps<T> {
-    if (renderItem) {
-      return renderItem({ item: card, index });
-    } else {
-      return <>{childrenArray[index]}</>;
-    }
-  };
+    const renderCard = (card: any, index: number): JSX.Element => {
+      if (renderItem) {
+        return renderItem({ item: card, index });
+      } else {
+        return <>{childrenArray[index]}</>;
+      }
+    };
 
-  const renderFirstCard = (): JSX.Element | undefined => {
-    if (cardsData.length) {
-      return renderCard(cardsData[0], 0);
-    }
-    return undefined;
-  };
-
-  const cardKeyExtractor = (card: any) => {
-    if (keyExtractor) {
-      return keyExtractor(card);
-    } else {
-      return card?.toString();
-    }
-  };
+    const renderFirstCard = (): JSX.Element | undefined => {
+      if (cardsData.length) {
+        return renderCard(cardsData[0], 0);
+      }
+      return undefined;
+    };
 
-  /* 
+    const cardKeyExtractor = (card: any) => {
+      if (keyExtractor) {
+        return keyExtractor(card);
+      } else {
+        return card?.toString();
+      }
+    };
+
+    /* 
   react-native-deck-swiper does not re-render cards when parent state changes
   This forces an update on every re-render to reflect any parent state changes
   */
-  React.useEffect(() => {
-    deckSwiperRef.current?.forceUpdate();
-  });
-
-  /**
-   * By default react-native-deck-swiper positions everything with absolute position.
-   * To overcome this, it is wrapped in a View to be able to add the component in any layout structure.
-   *
-   *
-   * Since all children of that View are absolutley positioned, the View does not have a height and still looks and behaves weird.
-   * To fix/mitage this without setting a static height, the first card is rendered in invisible state to take up space.
-   * This effectivley makes the default height of the container be the height of the first card.
-   */
-
-  return (
-    <View>
-      <View style={styles.containerHeightFiller}>{renderFirstCard()}</View>
-      <DeckSwiperComponent
-        ref={deckSwiperRef}
-        cards={cardsData as any[]}
-        renderCard={renderCard}
-        keyExtractor={cardKeyExtractor}
-        containerStyle={
-          StyleSheet.flatten([styles.cardsContainer, style]) as
-            | object
-            | undefined
+    React.useEffect(() => {
+      deckSwiperRef.current?.forceUpdate();
+    });
+
+    React.useImperativeHandle(ref, () => ({
+      swipeCard: (direction: "left" | "right" | "top" | "bottom") => {
+        switch (direction) {
+          case "left":
+            deckSwiperRef.current?.swipeLeft();
+            break;
+          case "right":
+            deckSwiperRef.current?.swipeRight();
+            break;
+          case "top":
+            deckSwiperRef.current?.swipeTop();
+            break;
+          case "bottom":
+            deckSwiperRef.current?.swipeBottom();
+            break;
+          default:
+            deckSwiperRef.current?.swipeLeft();
```

Comment:
Only thing that's new, alongside wrapping everything in a `forwardRef`. Rest is indentation changes